retrieval/target.go

```diff
@@ -361,6 +364,16 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	for samples := range t.ingestedSamples {
 		for _, s := range samples {
 			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
+			labels, err := Relabel(clientmodel.LabelSet(s.Metric), t.metricRelabelConfigs...)
```

Comment:
In `Relabel()`, this makes an extra copy of the entire labelset for every scraped sample, even if there are no metric relabeling configs. Maybe we could add an early return in https://github.com/prometheus/prometheus/blob/relabel-scrapes/retrieval/relabel.go#L16 if `len(cfgs) == 0`? (there or in the caller)